node/bft/src/sync/mod.rs

```diff
@@ -484,13 +484,46 @@ impl<N: Network> Sync<N> {
                         continue;
                     }
 
+                    if let Authority::Quorum(subdag) = block.authority() {
+                        // Retrieve the leader certificate of the subdag.
+                        let leader_certificate = subdag.leader_certificate();
+                        let leader_round = leader_certificate.round();
+                        let leader_author = leader_certificate.author();
+                        let leader_id = leader_certificate.id();
+
+                        // If a BFT sender was provided, commit the leader certificate. 
+                        if let Some(bft_sender) = self.bft_sender.get() {
+                            // Send the leader certificate to the BFT. 
+                            if let Err(e) = bft_sender.send_commit_bft(leader_certificate.clone()).await {
+                                bail!("Sync - {e}");
+                            };
+
+                            // Ensure that leader certificate was recently committed in the DAG. 
+                            match bft_sender.send_sync_is_recently_committed(leader_round, leader_id).await {
+                                Ok(is_recently_committed) => {
+                                    if !is_recently_committed {
+                                        bail!(
+                                            "Sync - Failed to advance blocks - leader certificate with author {leader_author} from round {leader_round} was not recently committed.",
+                                        );
+                                    }
+                                    debug!(
+                                        "Sync - Leader certificate with author {leader_author} from round {leader_round} was recently committed.",
+                                    );
```

Comment:
Maybe a nit but the boolean can be moved into the match statement you get a compile warning if missing a branch and has better readability IMO.

i.e:

```rust
Ok(true) => {
  debug!(
    Sync - Leader certificate with author {leader_author} from round {leader_round} was recently committed.",
  );
}
Ok(false) => {
  bail!(
    "Sync - Failed to advance blocks - leader certificate with author {leader_author} from round {leader_round} was not recently committed.",
  );
}
// rest of cases
```